core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -17,16 +17,28 @@
 
 package kafka.utils
 
-import org.apache.log4j.Logger
+import org.slf4j.{LoggerFactory, MarkerFactory}
+
+object Log4jControllerInstantiator {
+  try {
+    val log4jController = Class.forName("kafka.utils.Log4jController").asInstanceOf[Class[Object]]
+    val instance = log4jController.newInstance()
+    CoreUtils.registerMBean(instance, "kafka:type=kafka.Log4jController")
+  } catch {
+    case e: ClassNotFoundException => {}
+  }
+}
 
 trait Logging {
   val loggerName = this.getClass.getName
-  lazy val logger = Logger.getLogger(loggerName)
+  lazy val logger = LoggerFactory.getLogger(loggerName)
 
   protected var logIdent: String = null
 
+  val fatalMarker = MarkerFactory.getMarker("FATAL")
+
```

Comment:
What about this?

`object Log4jControllerHolder {
  val instance = tryCreateInstance

  def tryGetInstance(): Try[Object] = {
    instance
  }

  private def tryCreateInstance(): Try[Object] = {
    try {
      val log4jController = Class.forName("kafka.utils.Log4jController").asInstanceOf[Class[Object]]
      val instance = log4jController.newInstance()
      CoreUtils.registerMBean(instance, "kafka:type=kafka.Log4jController")
      Try(instance)
    } catch {
      case e: Throwable => {
        Try(e)
      }
    }
  }
}
/*...*/
Log4jControllerHolder.tryGetInstance`